app/Http/Controllers/PageController.php

```diff
@@ -42,4 +42,87 @@ public function getCreditsPage() {
             'extensions' => DB::table('site_extensions')->get(),
         ]);
     }
+
+    /**
+     * Shows the site calendar page.
+     *
+     * @return \Illuminate\Contracts\Support\Renderable
+     */
+    public function getCalendarPage() {
+        return view('pages.calendar');
+    }
+
+    /**
+     * Returns all events from news, sales and prompts formatted as json.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function getCalendarEvents() {
+        $start_at = request()->get('start');
+        $end_at = request()->get('end');
+
+        // get all prompts within time frame, if hide_before_start is set or hide_after_end is set, filter them out if they are not within the prompt time frame
+        $prompts = \App\Models\Prompt\Prompt::where('is_active', 1)->where('start_at', '>', $start_at)->where('end_at', '<', $end_at)
```

Comment:
Nit, and add `use App\Models\Prompt\Prompt;` at the top, and same goes for the other Models referenced this way in here.
```suggestion
        $prompts = Prompt::where('is_active', 1)->where('start_at', '>', $start_at)->where('end_at', '<', $end_at)
```